packager/packager.cc

```diff
@@ -598,6 +598,39 @@ class PackageInfoImpl final : public core::packages::PackageInfo {
                 return core::packages::StrictDependenciesLevel::Dag;
         }
     }
+
+    bool importsTransitively(const core::GlobalState &gs, const PackageInfo &otherPkg) const {
+        Timer timeit(gs.tracer(), "importsTransitively");
+
+        UnorderedSet<core::packages::MangledName> seen;
+        vector<core::packages::MangledName> toVisit;
+        toVisit.push_back(name.mangledName);
+
+        while (!toVisit.empty()) {
+            auto current = toVisit.back();
+            toVisit.pop_back();
+            if (seen.contains(current)) {
+                continue;
+            }
+            toVisit.push_back(current);
+
+            if (current == otherPkg.mangledName()) {
+                return true;
+            }
+
+            auto &info = PackageInfoImpl::from(gs.packageDB().getPackageInfo(current));
+
+            for (auto &import : info.importedPackageNames) {
+                if (import.type == core::packages::ImportType::Test ||
+                    !gs.packageDB().getPackageInfo(import.name.mangledName).exists()) {
+                    continue;
+                }
```

Comment:
It's probably worth calling out in the doc comment that this only implements `importsTransitively` for the non-test imports, because that's the only thing that's looked at for the sake of strict deps/layering checks.